plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftLaunchController.java

```diff
@@ -105,14 +109,31 @@ public void launch(ILaunchConfiguration configuration, String mode,
 			mode = currentMode;
 			throw e;
 		} finally {
-			checkServerState(beh, mode);
+			checkServerState(beh, currentMode, mode);
```

Comment:
So it seems to me currentMode is always null when restarting the server (whether it was in normal or in debug mode).

Maybe we need to carry around the last known mode of the server (stored in the behavior's shared data?)